pkg/ddl/backfilling_operators.go

```diff
@@ -83,29 +83,34 @@ type OperatorCtx struct {
 }
 
 // NewDistTaskOperatorCtx is used for adding index with dist framework.
-func NewDistTaskOperatorCtx(ctx context.Context, taskID, subtaskID int64) *OperatorCtx {
+func NewDistTaskOperatorCtx(
+	ctx context.Context,
+	taskID, subtaskID int64,
+) (*OperatorCtx, context.CancelFunc) {
 	opCtx, cancel := context.WithCancel(ctx)
-	opCtx = logutil.WithFields(opCtx, zap.Int64("task-id", taskID), zap.Int64("subtask-id", subtaskID))
+	opCtx = logutil.WithFields(opCtx,
+		zap.Int64("task-id", taskID),
+		zap.Int64("subtask-id", subtaskID))
 	return &OperatorCtx{
 		Context: opCtx,
 		cancel:  cancel,
-	}
+	}, cancel
```

Comment:
why not just use OperatorCtx.Cancel?

Reply:
the unused variable (`cancel`) can remind the caller. This PR fixes some functions that forget to `cancel`

Reply:
I doubt the `remind` part, coders can simple ignore it as OperatorCtx already has it

We should unify the place to call `cancel`, not scattering it all around

Reply:
The developers have forgotten to call `OperatorCtx.Cancel` at line 678 in pkg/ddl/backfilling.go (there's also a missing in unit test files), so I prefer to use golang's built-in detect to prevent future bug.

Reply:
static checks doesn't help those logic bugs, but ok anyway